rpcs3/util/sysinfo.cpp

```diff
@@ -414,14 +414,21 @@ std::string utils::get_firmware_version()
 
 		version = version.substr(start, end - start);
 
-		// Trim version
+		// Trim version (e.g. '04.8900' becomes '4.89')
 		const usz trim_start = version.find_first_not_of('0');
-		const usz trim_end = version.find_last_not_of('0');
+		const usz dot_pos = version.find_first_of('.');
+		usz trim_end = version.find_last_not_of('0');
 
-		if (trim_start == umax)
+		if (trim_start == umax || dot_pos == umax)
 		{
 			return {};
 		}
+		
+		// Try to keep the second 0 in the minor version (e.g. '04.9000' becomes '4.90' instead of '4.9')
+		if (end > trim_end && trim_end > dot_pos && (trim_end - dot_pos) == 1)
```

Comment:
This is overcompilcated, I think that the leading and trailing zeroes were actually there to help the code to read the version from fixed positions and are not there just to make our lives at rpcs3 harder.

Reply:
Like we trim the zeroes the wrong way, we should consider their position.